### DISCUSSION TITLE
Is it possible to use parameters in MetaLama

### DISCUSSION THREAD

**Discussion by WaldemarCichon** [metalama/Metalama#103] [Q&A] [answered]
My aspect would need a variable parameter for a field outside of the aspect method/property. Now I use the same field name for all cases but if this should work independently, I would need a parameter to inject this field name.

**Comment by svick** [comment, marked as answer]
Yes, it's possible. If I understand you correctly, the easiest way is to use attribute constructor parameter and access that inside the aspect.

Something like ([see it in Metalama Sandbox](https://try.metalama.net/#N4IgZiBcDaoHZRABQE4HsDmKCGBbAdAMYDOIANCIYgK7ECWcGABALICmALtgDZ7b4AxHLjYB3NCgDW+AILEADm0IdiAbgA6cWg2bsuvXPyF4xE6QGE0AEzYat9RkwDKAT2Ic2BAJIB5O5oYPFDgeJkJeYmImVExhTWBNJiSmeRQ6ADdsDyZ3LLpCJgAVNgAPDgB1NKCmAH1RKrYUJgBeJks4YjRuNnwfag5-OGSUtMzs3I58pnS0OitWbAYACgBKROSEoeHkjhQXde2mTcPDmSsrJaYARjJrphW7E+2zi6YABlur+8eTgF8D7aELKEAAWRyY-y2SUhAOgABlMEsQiI0GAlnUGigVisALoA1IZLJsHJcSYFQJMF6XCnYW4UgBG9wBxxOdDATEu2BarTeTKh2xZTySHBB6FETDgYkpKAw1BEcA4fUVYAASthGGwAKIlQhseSTNBwS7ItiopbYFbfAHDGH84YoTjUYJMLkAaiY9J+0M0tvk1Hp3Cm4WwkSYCIwMg4uzo9P6xMgTB86UaaRsehB1jkimU8XxoyJTAd2Cshu4LhJaUcNTAdDY3CsADkTINhn6A1Nw5Ho7GPEt3JXmDW642TGt+YLttXa-WmyIWkwhzPmwDba3-YGCmhkyhU8SrC5kfkAPyJ7e79PWVbM63JTJNRfzVoiLj4QrYGWcV8uRSCadWYi9GAs5sOiD7ASs+BODo3SrC2JwPvgABqPDUGwACE+CVHQHhwgwIHqCAF7zLkKAeFY+AEQ8mg3kkDocE6QzPvwMS6mwFxUfykK-OQICSIgHAgL8OK-EAA)):

```c#
public class LogAttribute : OverrideMethodAspect
{
    private readonly string _fieldName;

    public LogAttribute(string fieldName)
    {
        _fieldName = fieldName;
    }

    public override dynamic? OverrideMethod()
    {
        var field = meta.Target.Type.Fields.OfName(_fieldName).Single();

        // Use field.Value here to do what you need.

        return meta.Proceed();
    }
}
```

Usage:

```c#
[Log(nameof(_writer))]
```
